### PR TITLE
fix: bug when the name of subcategory is empty

### PR DESCRIPTION
**Pull request by dixidroid**
fixed a bug when the name of subcategory was empty in Discussions

### CODE REVIEW

**Review by HamzaIsrar12 — COMMENTED** [review]
I've asked Moiz to confirm the default string for us. Meanwhile, I believe it would be wise to minimize the usage of `topic.name` to zero, as it could potentially cause issues in other areas as well.

Regarding an alternative approach, using something like this  code[snippet](https://github.com/openedx/edx-app-android/blob/89d8d4ed4bc12495b4d8fa14a4246e36cb5bfbf1/OpenEdXMobile/src/main/java/org/edx/mobile/model/discussion/DiscussionTopic.java#L68) to fetch the topic title from one common place seems like a good idea. Thoughts?

**Comment by dixidroid** [comment]
@HamzaIsrar12 I changed where the topic name located. Check it please. Waiting for Moiz's confirmation...

File: discussion/src/main/java/org/openedx/discussion/domain/model/TopicsData.kt [hunk 1]
```diff
@@ -10,4 +15,10 @@ data class Topic(
...
     val threadListUrl: String,
     val children: List<Topic>
-)
\ No newline at end of file
+) {
+
+    @Composable
+    fun getTopicName(): String {
+        return name.ifEmpty { stringResource(id = R.string.discussion_unnamed_subcategory) }
+    }
```
[the remark this answers is not in the document]
Comment (HamzaIsrar12) [reply discussion/src/main/java/org/openedx/discussion/domain/model/TopicsData.kt:23]: Continuing your thought process, @volodymyr-chekyrta, perhaps we can update it at the repository level.

```
    suspend fun getCourseTopics(courseId: String): List<Topic> {
        val topicsData = api.getCourseTopics(courseId).mapToDomain()

        // Fetch default name
        val defaultTopicName = resourceManager.getString(R.string.discussion_unnamed_subcategory)

        currentCourseId = courseId
        topics.clear()
        topics.addAll(topicsData.nonCoursewareTopics)
        topicsData.coursewareTopics.forEach {
            topics.add(it)
            if (it.children.isNotEmpty()) {
                topics.addAll(it.children)
            }
        }
        
        // New Code
        topics.forEachIndexed { index, topic ->
            if (topic.name.isBlank()) {
                topics[index] = topic.copy(name = defaultTopicName)
            }
        }
        // End
        
        return topics.toList()
    }
```

Thoughts?

**Comment by HamzaIsrar12** [comment]
@dixidroid Moiz has [approved](https://github.com/openedx/openedx-app-android/issues/247#issuecomment-2006224315) the usage of "Unnamed subcategory".  👍🏻

File: discussion/src/main/java/org/openedx/discussion/domain/model/TopicsData.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (dixidroid) [reply discussion/src/main/java/org/openedx/discussion/domain/model/TopicsData.kt]: done

**Review by HamzaIsrar12 — APPROVED** [review]

**Pull request merged**